src/templates/create-modal-block/index.vue

```diff
@@ -1,3 +1,162 @@
+  const browseHeader = ref('browse-templates')
+  const recommendedHeader = ref('recommended-for-you')
+  const items = ref([
+    {
+      label: 'Recommended',
+      value: 'recommended'
+    },
+    {
+      label: 'Templates',
+      value: 'browse'
+    },
+    {
+      label: 'Resources',
+      value: 'new_resource'
+    }
+  ])
+  const resources = ref([
+    {
+      label: 'Domains',
+      to: '/domains/create'
+    },
+    {
+      label: 'Edge Application',
+      to: '/edge-applications/create'
+    },
+    {
+      label: 'Variables',
+      to: '/variables/create'
+    },
+    {
+      label: 'Intelligent DNS',
+      to: '/intelligent-dns/create'
+    },
+    {
+      label: 'Edge Firewall',
+      to: '/edge-firewall/create'
+    },
+    {
+      label: 'Edge Nodes',
+      to: '/edge-node/create'
+    },
+    {
+      label: 'Data Streaming',
+      to: '/data-streaming/create'
+    },
+    {
+      label: 'Edge Functions',
+      to: '/edge-functions/create'
+    },
+    {
+      label: 'Edge Services',
+      to: '/edge-services/create'
+    },
+    {
+      label: 'Digital Certificates',
+      to: '/digital-certificates/create'
+    },
+    {
+      label: 'Network Lists',
+      to: '/network-lists/create'
+    }
+  ])
+
+  const showRecommended = computed(() => {
+    return selectedTab.value === 'recommended'
+  })
+  const showBrowse = computed(() => {
+    return selectedTab.value === 'browse'
+  })
+  const showResource = computed(() => {
+    return selectedTab.value === 'new_resource'
+  })
+
+  const redirectToSolution = (template) => {
+    const params = {
+      vendor: template.vendor.slug,
+      solution: template.slug
+    }
+    router.push({ name: 'create-something-new', params })
+    emit('closeModal')
+  }
+
+  const redirect = (toLink) => {
+    router.push(toLink)
+    emit('closeModal')
+  }
+
+  const loadRecommendedSolutions = async () => {
+    try {
+      isLoading.value = true
+      const payload = { type: recommendedHeader.value }
+      templates.value = await MarketplaceService.listSolutionsService(payload)
+    } catch (error) {
+      toast.add({
+        closable: true,
+        severity: 'error',
+        summary: error
+      })
+    } finally {
+      isLoading.value = false
+    }
+  }
+
+  const loadBrowse = async () => {
+    try {
+      isLoading.value = true
+      const payload = { type: browseHeader.value }
+      browseTemplates.value = await MarketplaceService.listSolutionsService(payload)
+    } catch (error) {
+      toast.add({
+        closable: true,
+        severity: 'error',
+        summary: error
+      })
+    } finally {
+      isLoading.value = false
+    }
+  }
+
+  const onMenuChange = async (target) => {
+    if (target.value === null) {
+      selectedTabControl.value = selectedTab.value
+    } else {
+      selectedTab.value = target.value
+    }
+    if (target.value === 'browse') {
+      if (browseTemplates.value.length === 0) {
```

Comment:
Sugestão: talvez poderia colocar algo como:
```suggestion
      if (!browseTemplates.value.length) {
```